src/nvim/event/multiqueue.c

```diff
@@ -245,3 +252,33 @@ static MultiQueueItem *multiqueue_node_data(QUEUE *q)
 {
   return QUEUE_DATA(q, MultiQueueItem, node);
 }
+
+/// Allow an event to be processed by multiple child queues to the main queue
```

Comment:
@bfredl is it really limited to "child queues of the main queue" ? Seems like any queue could be used (safely), no?

Reply:
Sure, it is just easier to reason about. Events will always be processed by the main queue "eventually", child queues processes subsets of the main queue events. This allows events to be in two (or more) such named subsets.